server/cmwell-ws/app/security/EagerAuthCache.scala

```diff
@@ -64,13 +68,20 @@ class EagerAuthCache @Inject()(crudServiceFS: CRUDServiceFS)(implicit ec: Execut
     }
   }
 
-  def invalidate(nbg: Boolean): Unit = load()
+  def invalidate(nbg: Boolean): Future[Boolean] = if(isLoadingSemaphore) Future.successful(false) else load(nbg)
 
-  private def load(): Unit = {
-    retryUntil[AuthData](isSuccessful = !_.isEmpty, maxRetries = 30, delay = 10.seconds)(loadOnce(false)).map { data => oData = combiner(oData, data) }
-    delayedTask(5.seconds)(
-      retryUntil[AuthData](isSuccessful = !_.isEmpty, maxRetries = 30, delay = 10.seconds)(loadOnce(true)).map { data => nData = combiner(nData, data) }
-    )
+  private def load(nbg: Boolean): Future[Boolean] = {
+    isLoadingSemaphore = true
```

Comment:
where is it tested?
I'm expecting a: `if(isLoadingSemaphore) { /* do something */ }` somewhere...
Sill thinks `atomicBoolean.testAndSet` is better suited here, in case it's gonna be used that is.

Reply:
it is tested in the `invalidate` method.
Do we really need the AtomicBoolean here?

Reply:
why only protect against one side?
If you `invalidate()` a nanosec before scheduler is invoked, you'll get concurrent updates.
also, an abusive (malicious or buggy) user can `invalidate()` multiple times concurrently. you need to protect against that as well.

you named it "semaphore". either it is, or it's not. 
use an atomic boolean (shouldn't be called often anyway, overhead can't hurt),
or don't use anything at all.

at least that what I think. in any case, I'm okay with going with it as is, as long as we are aware it is not really protected, (inline docs?)